app/quest-boost/[boostId]/page.tsx

```diff
@@ -166,29 +103,31 @@ export default function Page({ params }: BoostQuestPageProps) {
             {participants} players
           </p>
         </div>
-        <div>
-          <Button
-            disabled={
-              boost?.claimed ||
-              hexToDecimal(boost?.winner ?? "") !== hexToDecimal(address)
-            }
-            onClick={handleClaimClick}
-          >
-            {(() => {
-              if (boost?.claimed) {
-                return "Claimed ✅";
-              } else if (
-                hexToDecimal(boost?.winner ?? "") === hexToDecimal(address)
-              ) {
-                return "Claim boost reward 🎉 ";
-              } else if (boost && boost?.expiry > Date.now()) {
-                return "Boost has not ended ⌛";
-              } else {
-                return "You’re not selected 🙁";
+        {address ? (
+          <div>
+            <Button
+              disabled={
+                boost?.claimed ||
+                hexToDecimal(boost?.winner ?? "") !== hexToDecimal(address)
               }
-            })()}
-          </Button>
-        </div>
+              onClick={() => router.push(`/quest-boost/claim/${boost?.id}`)}
+            >
+              {(() => {
+                if (boost?.claimed) {
+                  return "Claimed ✅";
+                } else if (
+                  hexToDecimal(boost?.winner ?? "") === hexToDecimal(address)
+                ) {
+                  return "Claim boost reward 🎉 ";
+                } else if (boost && boost?.expiry > Date.now()) {
+                  return "Boost has not ended ⌛";
+                } else {
+                  return "You’re not selected 🙁";
+                }
+              })()}
```

Comment:
Ohh it's ugly 😭

Please use a function haha

Reply:
done